src/py/mat3ra/made/tools/build/defect/builders.py

```diff
@@ -371,3 +378,143 @@ def _generate(self, configuration: _ConfigurationType) -> List[_GeneratedItemTyp
+        """
+        if material.basis.is_in_cartesian_units:
+            crystal_basis = material.basis.copy()
+            crystal_basis.to_crystal()
+            material.basis = crystal_basis
+        atoms = to_ase(material)
+        atoms.rotate(v=axis, a=angle, center="COM")
+        atoms.wrap()
+
+        return Material(from_ase(atoms))
+
+    def create_terrace(
+        self,
+        material: Material,
+        cut_direction: Optional[List[int]] = None,
+        pivot_coordinate: Optional[List[float]] = None,
+        steps_number: int = 1,
+        use_cartesian_coordinates: bool = False,
+        rotate_to_match_pbc: bool = True,
+    ) -> List[Material]:
+        """
+        Create a terrace at the specified position on the surface of the material.
+
+        Args:
+            material: The material to add the terrace to.
+            cut_direction: The direction of the cut in lattice directions.
+            pivot_coordinate: The center position of the terrace.
+            steps_number: The number of steps to bunch.
+            use_cartesian_coordinates: Whether to use Cartesian coordinates for the center position.
+            rotate_to_match_pbc: Whether to rotate the material to match the periodic boundary conditions.
+        Returns:
+            The material with the terrace added.
+        """
+        if cut_direction is None:
+            cut_direction = [0, 0, 1]
+        if pivot_coordinate is None:
+            pivot_coordinate = [0.5, 0.5, 0.5]
+
+        new_material = material.clone()
+        material_with_additional_layers = self.create_material_with_additional_layers(new_material, steps_number)
+
+        normalized_direction_vector = self._calculate_cut_direction_vector(material, cut_direction)
+        condition = CoordinateConditionBuilder().plane(
+            plane_normal=normalized_direction_vector,
+            plane_point_coordinate=pivot_coordinate,
+        )[0]
+        atoms_within_terrace = filter_by_condition_on_coordinates(
+            material=material_with_additional_layers,
+            condition=condition,
+            use_cartesian_coordinates=use_cartesian_coordinates,
+        )
+        merged_material = self.merge_slab_and_defect(new_material, atoms_within_terrace)
+
+        angle, normalized_rotation_axis, delta_length = self._calculate_rotation_parameters(
+            material, merged_material, normalized_direction_vector
+        )
+        result_material = translate_to_z_level(merged_material, "center")
+
+        if rotate_to_match_pbc:
+            adjusted_material = self._increase_lattice_size(result_material, delta_length, normalized_direction_vector)
+            result_material = self._rotate_material(
+                material=adjusted_material, axis=normalized_rotation_axis, angle=angle
+            )
+        return [result_material]
+
+    def _generate(self, configuration: _ConfigurationType) -> List[_GeneratedItemType]:
+        return self.create_terrace(
```

Comment:
self.create_terrace should return a single material and here you can create a 1-element array instead. Same for `create_island`